src/phoenix/experimental/evals/models/openai.py

```diff
@@ -24,16 +24,44 @@
 logger = logging.getLogger(__name__)
 
 
+AZURE_REQUIRED_OPTIONS = ["api_version", "azure_endpoint"]
+AZURE_ADDITIONAL_OPTIONS = [
+    "azure_endpoint",
+    "azure_deployment",
+    "azure_ad_token",
+    "azure_ad_token_provider",
+]
+
+
+@dataclass
+class AzureOptions:
+    api_version: str
+    azure_endpoint: str
+    azure_deployment: Optional[str]
+    azure_ad_token: Optional[str]
+    azure_ad_token_provider: Optional[Callable[[], str]]
+
+
 @dataclass
 class OpenAIModel(BaseEvalModel):
-    openai_api_type: Optional[str] = field(default=None)
-    openai_api_version: Optional[str] = field(default=None)
-    openai_api_key: Optional[str] = field(repr=False, default=None)
-    openai_organization: Optional[str] = field(repr=False, default=None)
-    engine: str = ""
-    """Azure engine (the Deployment Name of your model)"""
+    # openai_api_type: Optional[str] = field(default=None)
+    api_key: Optional[str] = field(repr=False, default=None)
+    """Your OpenAI key. If not provided, will be read from the environment variable"""
+    organization: Optional[str] = field(repr=False, default=None)
+    """
+    The organization to use for the OpenAI API. If not provided, will default
+    to what's configured in OpenAI
+    """
+    # Azure options
+    api_version: Optional[str] = field(default=None)
+    """https://learn.microsoft.com/en-us/azure/ai-services/openai/reference#rest-api-versioning"""
+    azure_endpoint: Optional[str] = field(default=None)
```

Comment:
Maybe we should leave this as `endpoint` for the use case where people deploy a local server that pretends to be OpenAI (but running llama-2 or something)?
```suggestion
    endpoint: Optional[str] = field(default=None)
```

Reply:
Interesting. Hadn't thought of that. On one hand I like how explicit and 1-to-1 this is but I see your point. Will change.

Reply:
Actually I'm gonna split it since I'm not 100% confident that we should conflate the two in case someone puts a load balancer in front of the endpoint.